### PR TITLE
Content tweaks for NISRA (typo, school address, quals route)

### PR DESCRIPTION
**Pull request by bambrp**
### What is the context of this PR?
* 1 year ago address - current response option "Student term time in the UK" change to Student term time address in the UK
* Routing wrong on quals- if any yes answers to nvq, gcse, a-levels, degree, skip other-quals and onto apprenticeship. Otherwise show other-quals then apprenticeship.
* Missing two questions https://overflow.io/s/30TRR3/?node=739ef6db & https://overflow.io/s/30TRR3/?node=21a93787 which is the place of study and how to travel to school for under 16s

I've also made a change to the Over16 lib test, changing it to "less than or equal", to capture people who have turned 16 that day- this does bring up a corner case of time zones, but this is hopefully a better, not perfect, option.

Had to de-share the carer block, due to it not longer having the same routing rules across all regions.

### How to review 
Build schema, run through and test.

### CODE REVIEW

**Comment by berroar** [comment]
I built the schema and began to run through to test, but I noticed that the place of study and method of travel to school questions appear regardless of the age that I enter.

**Comment by berroar** [comment]
Not related to the changes here - but while testing I noticed that for the for the 'arrive-in-country' question, there is no validation to stop you from being able to enter a date that is before your own date of birth.

**Review by berroar — APPROVED** [review]
Successfully built schema and tested the changes. All appears to be OK.

File: data-source/jsonnet/northern-ireland/census_individual.jsonnet [hunk 1]
```diff
@@ -151,6 +155,14 @@ function(region_code) {
...
           ],
         },
+        {
+          id: 'school-group',
+          title: 'School',
+          blocks: [
+            school_location,
+            school_travel_mode,
+          ],
+        },
```
Comment (ajmaddaford) [inline data-source/jsonnet/northern-ireland/census_individual.jsonnet]: This group seems a bit unnecessary, maybe we could rename `Qualifications` to `Education` and merge these blocks into that group?

**Review by ajmaddaford — DISMISSED** [review]

File: data-source/jsonnet/northern-ireland/census_individual.jsonnet (hunk 1 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/census_individual.jsonnet]: As it is reached from multiple groups, and we can't deep link into groups, this seems necessary.

File: data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet [hunk 2]
```diff
@@ -1,9 +1,9 @@
 local placeholders = import '../../../../common/lib/placeholders.libsonnet';
 local rules = import '../../../../common/lib/rules.libsonnet';
 
-local nonProxyTitle = 'Have you achieved an AS, A level or equivalent qualification?';
+local nonProxyTitle = 'Have you achieved an A level, AS level or equivalent qualification?';
```
Comment (MebinAbraham) [inline data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet]:
```suggestion
local nonProxyTitle = 'Have you achieved an A level, AS level or equivalent qualifications?';
```

File: data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 local proxyTitle = {
-  text: 'Has <em>{person_name}</em> achieved an AS, A level or equivalent qualification?',
+  text: 'Has <em>{person_name}</em> achieved an A level, AS level or equivalent qualification?',
```
Comment (MebinAbraham) [inline data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet]:
```suggestion
  text: 'Has <em>{person_name}</em> achieved an A level, AS level or equivalent qualifications?',
```

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/past_usual_household_address.jsonnet [hunk 4]
```diff
@@ -81,6 +81,18 @@ local proxyDescription = 'If they had no usual address one year ago, state the a
...
       },
     },
+    {
+      goto: {
+        block: 'last-year-address',
+        when: [
+          {
+            id: 'past-usual-address-household-answer',
+            condition: 'equals',
+            value: 'Student term time address in the UK',
```
Comment (MebinAbraham) [inline data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/past_usual_household_address.jsonnet]: You could remove this extra block and use the `equals any` rule.
```jsonnet
{
      goto: {
        block: 'last-year-address',
        when: [
          {
            id: 'past-usual-address-household-answer',
            condition: 'equals any',
            values: [
              'Another address in the UK',
              'Student term time address in the UK',
            ],
          },
        ],
      },
    },
```

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/religion.jsonnet [hunk 5]
```diff
@@ -73,8 +81,8 @@ local question(title) = {
         block: 'no-religion',
         when: [
           {
-            id: 'religion-answer',
-            condition: 'equals',
+            id: 'religion-answer-exclusive',
+            condition: 'contains',
```
Comment (MebinAbraham) [inline data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/religion.jsonnet]: Up to you, if you use `condition: 'set',` and remove the `value` line then it would work the same way and it's 1 less place to change content that could be missed.

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/sexual_identity.jsonnet [hunk 6]
```diff
@@ -59,12 +56,19 @@ local proxyTitle = {
...
     {
-      question: question(proxyTitle, proxyDetailAnswerLabel),
+      question: question(proxyTitle),
       when: [rules.proxyYes],
     },
   ],
+  routing_rules: [
+    {
+      goto: {
+        group: 'qualifications-group',
```
Comment (MebinAbraham) [inline data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/sexual_identity.jsonnet]: Is this routing rule needed as it would go to this group anyway?

File: data-source/jsonnet/northern-ireland/blocks/individual/school/school_travel_mode.jsonnet [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    {
+      question: question(proxyTitle, proxyGuidance),
+      when: [rules.proxyYes],
+    },
+  ],
+  routing_rules: [
+    {
+      goto: {
+        group: 'comments-group',
```
Comment (MebinAbraham) [inline data-source/jsonnet/northern-ireland/blocks/individual/school/school_travel_mode.jsonnet]: Is this needed?. This block is followed by `comments-group`.

**Review by MebinAbraham — CHANGES_REQUESTED** [review]
Some minor things.

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/religion.jsonnet (hunk 5 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/religion.jsonnet]: Given how we're using this mutually exclusive (the only option in that second group is the no/none option), that would work, yes, good idea.

File: data-source/jsonnet/northern-ireland/blocks/individual/school/school_travel_mode.jsonnet (hunk 7 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/school/school_travel_mode.jsonnet]: It was sort of insurance against this group being moved, IIRC.

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/sexual_identity.jsonnet (hunk 6 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/sexual_identity.jsonnet]: It was sort of insurance against this group being moved, IIRC.

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/past_usual_household_address.jsonnet (hunk 4 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/past_usual_household_address.jsonnet]: Another good idea, I didn't know about equals any

File: data-source/jsonnet/northern-ireland/blocks/individual/school/school_travel_mode.jsonnet (hunk 7 above)
Comment (bitdivision) [reply data-source/jsonnet/northern-ireland/blocks/individual/school/school_travel_mode.jsonnet]: We should probably be consistent about this and get rid of it or enforce it.

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/past_usual_household_address.jsonnet (hunk 4 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/past_usual_household_address.jsonnet]: done

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/sexual_identity.jsonnet (hunk 6 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/sexual_identity.jsonnet]: Decided to remove. It adds a different risk, but it's the one with less code.

File: data-source/jsonnet/northern-ireland/blocks/individual/school/school_travel_mode.jsonnet (hunk 7 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/school/school_travel_mode.jsonnet]: Decided to remove. It adds a different risk, but it's the one with less code.

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/religion.jsonnet (hunk 5 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/religion.jsonnet]: done

File: data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet [hunk 8]
```diff
@@ -1,9 +1,9 @@
 local placeholders = import '../../../../common/lib/placeholders.libsonnet';
 local rules = import '../../../../common/lib/rules.libsonnet';
 
-local nonProxyTitle = 'Have you achieved an AS, A level or equivalent qualification?';
+local nonProxyTitle = 'Have you achieved an A level, AS level or equivalent qualifications?';
```
Comment (iwootten) [inline data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet:4]: I don't understand why this/subsequent questions have changed to 'qualifications' (plural)? Were they originally incorrectly worded? It no longer seems correct as neither an a level or as level is more than a single qualification.

Comment (MebinAbraham) [reply data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet:4]: 🤷‍♂ I pointed it out only cause card states `qualifications`.

Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet:4]: It feels a bit off, as the preceding items are singular. But it could be argued that it may be- as it is in the case of an AS level, that you might need more than one qual to form equivalence i.e. two AS levels for one A level.

The or creates a break in the sentence allowing for a plurality disagreement, I *think*.

But it is what was requested, as Mebin correctly noticed.

Comment (iwootten) [reply data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet:4]: Ah, fair enough - difficult to see which card this relates to given its not linked to here.

**Review by MebinAbraham — DISMISSED** [review]

File: data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet (hunk 8 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/qualifications/a_level.jsonnet:4]: Ah sorry, thought most people came via Trello. https://trello.com/c/JrPo6wKV/3034-nisra-content-tweaks

**Review by MebinAbraham — APPROVED** [review]

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/national_identity.jsonnet [hunk 9]
```diff
@@ -9,37 +9,22 @@ local proxyTitle = {
...
-
 local nonProxyDetailAnswerLabel = 'Please describe your national identity';
 local proxyDetailAnswerLabel = 'Please describe their national identity';
 
-local question(title, definitionContent, detailAnswerLabel) = {
+local question(title, detailAnswerLabel) = {
   id: 'national-identity-question',
   title: title,
   type: 'General',
+<<<<<<< HEAD
```
Comment (iwootten) [inline data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/national_identity.jsonnet]: Looks like this has been badly merged?

**Review by iwootten — CHANGES_REQUESTED** [review]

File: data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/national_identity.jsonnet (hunk 9 above)
Comment (bambrp) [reply data-source/jsonnet/northern-ireland/blocks/individual/identity-and-health/national_identity.jsonnet]: Yes, of course it has. Grr. File was still open in the editor after I'd made the changes.

**Review by iwootten — APPROVED** [review]

**Comment by bambrp** [comment]
Nudge

**Comment by bambrp** [comment]
Bump
